examples/applications/restapi/tests/test_restapi_handlers.py

```diff
@@ -417,31 +422,86 @@ async def test_get_balance_good_response(self, monkeypatch, cli):
         assert resp.status == 200
         response = await resp.read()
         assert json.loads(response) == expected_json
 
     async def test_remove_txs_specific_txid(self, monkeypatch, cli):
-        monkeypatch.setattr(self.rest_server, 'remove_signed_transaction',
-                            _fake_remove_signed_transaction)
+        monkeypatch.setattr(self.rest_server, 'remove_transaction',
+                            _fake_remove_transaction)
+
+        expected_response = {
+                "items": [
+                    {
+                        'id': '0000000000000000000000000000000000000000000000000000000000000000',
+                        'result': 200
+                     }
+            ]
+        }
 
         # mock request
         network = "test"
         wallet_name = "wallet_file1.sqlite"
         account_id = "1"
         txids = ["00" * 32]
-        resp = await cli.post(f"/v1/{network}/dapp/wallets/{wallet_name}/"
+        resp = await cli.delete(f"/v1/{network}/dapp/wallets/{wallet_name}/"
                               f"{account_id}/txs/remove",
                               data=json.dumps({"txids": txids}))
 
-        assert resp.status == 200, await resp.read()
+        assert resp.status == 207, await resp.read()
+        response = await resp.read()
+        assert json.loads(response) == expected_response
+
+    async def test_remove_txs_specific_txid_failed_to_delete(self, monkeypatch, cli):
+        monkeypatch.setattr(self.rest_server, 'remove_transaction',
+                            _fake_remove_transaction_raise_fault)
+
+        expected_response = {
+                "items": [
+                    {
+                        'id': '0000000000000000000000000000000000000000000000000000000000000000',
+                        'result': 400,
+                        'description': 'DisabledFeatureError: You used this endpoint in a way that '
+                                       'is not supported for safety reasons. See documentation for '
+                                       'details (https://electrumsv.readthedocs.io/ )',
+                    }
+            ]
+        }
+
+        # mock request
+        network = "test"
+        wallet_name = "wallet_file1.sqlite"
+        account_id = "1"
+        txids = ["00" * 32]
+        resp = await cli.delete(f"/v1/{network}/dapp/wallets/{wallet_name}/"
+                              f"{account_id}/txs/remove",
+                              data=json.dumps({"txids": txids}))
+
+        assert resp.status == 207, await resp.read()
+        response = await resp.read()
+        assert json.loads(response) == expected_response
+
+    async def test_remove_txs_bad_request(self, monkeypatch, cli):
+        monkeypatch.setattr(self.rest_server, 'remove_transaction',
+                            _fake_remove_transaction_raise_fault)
+
+        expected_response = \
+            {'code': 40403, 'message': "Required body variable: 'txids' was not provided."}
```

Comment:
What is the value of having a code for the message?

Reply:
Well it doesn't cost me anything because I prefer to do `if` statements using the code/integer value of the `Fault` rather than string parsing... 

And maybe the user would want to parse the error codes as well?

I don't feel strongly about it though. Happy to ditch... but the http 400 error doesn't provide any way of differentiating between ~ 15 types of existing 400 errors that can be encountered by use of the restapi...

Reply:
I should have been more specific. The question in this case is whether a missing required variable needs a code, because there is no value in the caller reacting to it besides propagating the message to it's own logs. It can't retry, it can't do anything. A 500 error with a more specific message, if possible, is probably as worthwhile. This does not mean there aren't errors for which there are codes that help, but more commenting on this case and cases like it.

Reply:
Oh I see... I think there should be some kind of code + message fields to make parsing responses uniform. But maybe you'd like it better if we removed:

    `HEADER_VAR_NOT_PROVIDED_CODE = 40402` and
    `BODY_VAR_NOT_PROVIDED_CODE = 40403`

If you prefer and use

    `GENERIC_BAD_REQUEST_CODE = 40000` instead

Reply:
What I am saying is that what reason is there to parse the message if the code cannot do anything about it? The message is therefore for the user.

Reply:
I agree with you on that. It's purely for the human to read. But often times the user will write code expecting a code + message in case of a non-200 OK http response. So imo we should still have those fields present in some shape or form. yes?

Reply:
I don't know. Make whatever call you think makes sense, you've spent more time working on this than me, so go with your preference.